test/conftest.py

```diff
@@ -85,23 +91,51 @@ def patch_logging(moulinette):
         },
     }
 
-    moulinette.init(
-        logging_config=logging,
-        _from_source=False
-    )
-
 
 @pytest.fixture(scope='session', autouse=True)
-def moulinette():
+def moulinette(tmp_path_factory):
     import moulinette
 
+    # Can't call the namespace just 'test' because
+    # that would lead to some "import test" not importing the right stuff
+    namespace = "moulitest"
+    tmp_cache = str(tmp_path_factory.mktemp("cache"))
+    tmp_data = str(tmp_path_factory.mktemp("data"))
+    tmp_lib = str(tmp_path_factory.mktemp("lib"))
+    os.environ['MOULINETTE_CACHE_DIR'] = tmp_cache
```

Comment:
I would recommend to use the `monkeypatch` fixture for patching the environment. Please see https://docs.pytest.org/en/3.0.1/monkeypatch.html. It means that pytest will take care to clean up any changes to the environment for you. Can stop crazy hard to find env bugs later on for someone else.

Reply:
Many thanks for that pointer, looks like the info I was missing to do this cleanly, will have a deeper look

Reply:
Got reworked in https://github.com/YunoHost/moulinette/pull/216/commits/53ab4709d167ca70b4288a36706de33b1bb60221